src/libbson/src/bson/bson-string.c

```diff
@@ -61,16 +62,23 @@ bson_string_t *
 bson_string_new (const char *str) /* IN */
 {
    bson_string_t *ret;
+   size_t len_sz;
 
    ret = bson_malloc0 (sizeof *ret);
-   ret->len = str ? (int) strlen (str) : 0;
+   if (str) {
+      len_sz = strlen (str);
+      BSON_ASSERT (bson_in_range_unsigned (uint32_t, len_sz + 1));
```

Comment:
If `len_sz` is `SIZE_MAX`, then `len_sz + 1` may overflow. Suggest changing to:

```suggestion
      BSON_ASSERT (len_sz <= UINT32_MAX - 1);
```